Add contribution guidelines in `CONTRIBUTING.md` (#4)
refunct-community/rx-backend: CONTRIBUTING.md

* Excludes all unnecessary types, methods, code blocks, source files, nuget dependencies and project configurations.
* Contains documentation or code comments illustrating expected vs. actual behavior.
* If possible, avoids performing any unneeded IO or system calls.

## Contributing Changes
Project maintainers will merge changes that improve rx-backend in any way and align with the goal of the project.

Maintainers will not merge changes that have narrowly-defined benefits, due to risk of compatibility and bloat. There are hundreds of speedrunners who rely on the functionality of rx-backend. Changes may be reverted if they are found to be breaking.

### DOs and DON'Ts
Please:
* **DO** follow the coding style (Visual Studio and the .editorconfig should be of help).
* **DO** give priority to the current style of the project or file you're changing even if it diverges from the general guidelines.
* **DO** keep discussions focused. When a new or adjacent topic comes up in an existing issue, it is often better to create a new issue than to side-track the current one.
* **DO** run `dotnet format`, followed by another build of the solution before committing your changes.

Please:

> * **DO NOT** create any pull requests without prior to filing an issue, unless the changes are trivial. Maintainers and other contributors will want to talk about the changes you want to make and come to an agreement.

Reviewer: `without prior to` is redundant

Author: "Do not create any pull requests filing an issue [...]"?

Reviewer 2: I think he means to omit the 'prior to' or 'without'.

   * Branches are useful since they isolate your changes from incoming changes from upstream. They also enable you to create multiple PRs from the same fork.
4. Apply and commit your changes to your branch.
5. Build the solution with your changes.
   * Before building, it is recommended that you run `dotnet format` in the directory of the solution.
   * Run all unit and integration tests and make sure they pass.
6. Create a pull request (PR) against the **`main`** branch.
   * In the PR description, state what issue or improvement your change is addressing.
   * Verify that all Continuous Integration checks are passing.
7. Wait for feedback or approval of your changes from maintainers, contributors, or community members.
8. If all checks pass and there is no negative feedback on your changes, your PR will be merged.
   * You can safely delete the branch and fork used for making the changes.

## Help Wanted
Issues marked with this label are currently not possible to be worked on due to a lack of knowledge on the maintainers' part. Please, if you have ideas to share, do not hesitate to add a comment on the issue.

## License Notice

> All code contributed to rx-backend falls under either of
> 
> * Apache License, Version 2.0, ([LICENSE-APACHE.md](./LICENSE-APACHE.md) or http://apache.org/licenses/LICENSE-2.0)
> * MIT license ([LICENSE-MIT.md](./LICENSE-MIT.md) or http://opensource.org/licenses/MIT)
> 
> at your choice.

Reviewer: When contributing, the code must fall under both licenses. Otherwise I could choose to contribute code under MIT but not under Apache. If someone then decided to use the code as Apache2, it would be in breach.

```suggestion
All code contributed to rx-backend falls under both

* Apache License, Version 2.0, ([LICENSE-APACHE.md](./LICENSE-APACHE.md) or http://apache.org/licenses/LICENSE-2.0)
* MIT license ([LICENSE-MIT.md](./LICENSE-MIT.md) or http://opensource.org/licenses/MIT)
```
